src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.study.server.networkmodificationtree;
+
+import org.gridsuite.study.server.StudyException;
+import org.gridsuite.study.server.networkmodificationtree.dto.AbstractNode;
+import org.gridsuite.study.server.networkmodificationtree.entities.AbstractNodeInfoEntity;
+import org.gridsuite.study.server.networkmodificationtree.repositories.NodeInfoRepository;
+import org.springframework.data.jpa.repository.Modifying;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * @author Jacques Borsenberger <jacques.borsenberger at rte-france.com>
+ */
+public abstract class AbstractNodeRepositoryProxy<NodeInfoEntity extends AbstractNodeInfoEntity, NodeInfoEntityRepository extends NodeInfoRepository<NodeInfoEntity>, NodeDto extends AbstractNode> {
+
+    final NodeInfoEntityRepository nodeInfoRepository;
+
+    protected AbstractNodeRepositoryProxy(NodeInfoEntityRepository nodeInfoRepository) {
+        this.nodeInfoRepository = nodeInfoRepository;
+    }
+
+    public abstract NodeInfoEntity toEntity(AbstractNode node);
+
+    public abstract NodeDto toDto(NodeInfoEntity node);
+
+    public void createNodeInfo(AbstractNode nodeInfo) {
+        nodeInfoRepository.save(toEntity(nodeInfo));
+    }
+
+    public void deleteByNodeId(UUID id) {
+        nodeInfoRepository.deleteById(id);
+    }
+
+    public NodeDto getNode(UUID id) {
+        return toDto(nodeInfoRepository.findById(id).orElseThrow(() -> new StudyException(StudyException.Type.ELEMENT_NOT_FOUND)));
+    }
+
+    protected NodeDto completeNodeInfo(AbstractNodeInfoEntity nodeEntity, NodeDto node) {
+        node.setId(nodeEntity.getNode().getIdNode());
+        node.setName(nodeEntity.getName());
+        node.setDescription(nodeEntity.getDescription());
+        return node;
+    }
+
+    protected NodeInfoEntity completeEntityNodeInfo(AbstractNode node, NodeInfoEntity entity) {
+        entity.setIdNode(node.getId());
+        entity.setName(node.getName());
+        entity.setDescription(node.getDescription());
+        return entity;
+    }
+
+    @Modifying
```

Comment:
https://docs.spring.io/spring-data/data-jpa/docs/current/api/org/springframework/data/jpa/repository/Modifying.html
"This annotation is only considered if used on query methods defined through a Query annotation"

Does this have an effect ?